metric_tank/in/in.go

```diff
@@ -0,0 +1,76 @@
+package in
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/grafana/grafana/pkg/log"
+	"github.com/raintank/met"
+	"github.com/raintank/raintank-metric/metric_tank/defcache"
+	"github.com/raintank/raintank-metric/metric_tank/mdata"
+	"github.com/raintank/raintank-metric/metric_tank/usage"
+	"github.com/raintank/raintank-metric/msg"
+	"github.com/raintank/raintank-metric/schema"
+)
+
+// In is a base handler for a metrics packet, aimed to be embedded by concrete implementations
+type In struct {
+	metricsPerMessage met.Meter
+	metricsReceived   met.Count
+	msgsAge           met.Meter // in ms
+	tmp               msg.MetricData
+
+	metrics  mdata.Metrics
+	defCache *defcache.DefCache
+	usage    *usage.Usage
+}
+
+func New(metrics mdata.Metrics, defCache *defcache.DefCache, usage *usage.Usage, input string, stats met.Backend) In {
+	return In{
+		metricsPerMessage: stats.NewMeter(fmt.Sprintf("%s.metrics_per_message", input), 0),
+		metricsReceived:   stats.NewCount(fmt.Sprintf("%s.metrics_received", input)),
+		msgsAge:           stats.NewMeter(fmt.Sprintf("%s.message_age", input), 0),
+		tmp:               msg.MetricData{Metrics: make([]*schema.MetricData, 1)},
+
+		metrics:  metrics,
+		defCache: defCache,
+		usage:    usage,
+	}
+}
+
+func (in In) Handle(data []byte) {
```

Comment:
Each message sent to kafka will be a single metric, it wont be a batch of metrics as is done with NSQ.